### DISCUSSION TITLE
Do we have to run qterm in the terminal everytime

### DISCUSSION THREAD

**Discussion by somtooo** [aws/codewhisperer-command-line-discussions#304] [General]
After installing Amazon q for terminal on wezterm i realized that i have to run qterm which runs a process, before i get access to autocompletion. The documentation makes it seem like it will just work out of the box

**Comment by brandonskiser** [comment]
You shouldn't have to run qterm manually. Do you have the integrations installed? You can check by running `q integrations status dotfiles` and `q integrations install dotfiles`.

Also in general running `q doctor` should ideally resolve any other issues with autocomplete.

**Reply by el-zoltan** [reply]
@brandonskiser, I'm getting this behavior where I have to run `qterm` before autocomplete starts working in the VS Code integrated terminal, but it's working fine in iTerm. Both are using my same `.zshrc` which does have the integration lines at the top and bottom.

**Comment by alanwilter** [comment]
Well, what happened for me is this:

- If ssh from my Mac to my remote server (Linux), Q works fine
- If I use the terminal on vscode-remote for this remote server, I need to manually run qterm.

However, even after `qterm`, if I do `git push` (on vscode-remote terminal) I get:
```
error: Q_PARENT env variable not set
```

**Comment by alanwilter** [comment]
Interestingly... I ssh to my remote linux and get the Q_PARENT value, then on vscode-remote terminal I do `export Q_PARENT=xxxxx` and then `qterm` and autocompletion works beautifully inside vscode-remote!
